text/0000-dot-profile-utility-buildpack.md

```diff
@@ -0,0 +1,131 @@
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md) resolves to remove shell-specific logic from the CNB Specification.
+Part of this includes removing support for `.profile` script in a future version of the Platform API.
+RFC 93 recommends supporting the `.profile` script functionality in a utility buildpack to avoid regressions.
+This proposal is to develop and support the `.profile` utility buildpack, allowing RFC 93 to be implemented without regression.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The target persona is a platform operator or implementor who wants to update to the latest platform API, while maintaining the `.profile` functionality for application developers.
+
+We propose developing and supporting a buildpack to provide an identical interface to the existing `.profile` functionality.
+It will:
+
+- Detect a `.profile` file in the app dir
+- Wrap the `.profile` file so that it implements the `exec.d` interface
+- Add the `exec.d` executable to the `<layer>/exec.d` directory so the launcher will apply it
+
+We propose the ID `buildpacksio/profie` for this buildpack (cf. [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md#what-it-is)).
+
+## Example 1 (environment variables)
+
+Here is an example of a `.profile` script, inspired by paketo-buidpacks/node-engine:
+
+```
+memory_in_bytes="$(cat /sys/fs/cgroup/memory/memory.limit_in_bytes)"
+MEMORY_AVAILABLE="$(( $memory_in_bytes / ( 1024 * 1024 ) ))"
+export MEMORY_AVAILABLE
+```
+
+The wrapper should ensure that the `MEMORY_AVAILABLE` environment variable is set in the environment with the proper value.
+
+## Example 2 (file system side effects)
+
+With this contrived `.profile` script:
+
+```
+echo 'hello' >> "$HOME/hello"
+```
+
+The wrapper would not need to set any environment varibales, but should maintain the side effect of creating the `$HOME/hello` file.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+In short, the wrapper should do:
+
+```
+#!/bin/bash
+set -euo pipefail
+source .profile
+env >&3
+```
+
+We acknoledge that the `env` output will need to be properly quoted to be valid TOML.
+We will use or create a tool to handle that formatting.
+
+This will write the environment, including any variables set in `.profile`, to the [`exec.d` output TOML](https://github.com/buildpacks/spec/blob/main/buildpack.md#execd-output-toml).
+Since it also executes the `.profile` script, any side effects will happen.
+This will solve for both of the simple examples above.
+
+On Windows, for `.profile.bat` scripts, we can take the same approach of wrapping the script like:
+
+```
+call .profile.bat
+set >&%CNB_EXEC_D_HANDLE%
+```
+
+# Migration
+[migration]: #migration
+
+This buildpack is net new, so has no inherent migration considerations.
```

Comment:
```suggestion
This buildpack is not new, so has no inherent migration considerations.
```

Reply:
Updated to just "new," which was what was intended.